src/components/config_parser.cpp

```diff
@@ -193,12 +193,16 @@ line_type config_parser::get_line_type(const string& line) {
     case '#':
       return line_type::COMMENT;
 
-    default:
+    default: {
+      if (string_util::contains(line, "﻿"))
```

Comment:
This will not detect the BOM in header or comment lines. A better place to put this check would be at the beginning of the `parse_line` method.

Please also don't put a literal BOM here because this looks like an empty string on github and probably many editors (BOM inside the text signifies a zero-width nonbreaking space). It's probably best to put a string constant that is constructed from a `char` array in the header file.

Reply:
I added the unit test and moved the check in dd65b76. I kept getting this redefined error when I tried to extract the BOM reference to a string constant in the `config_parser.hpp` file, I assume that somewhere the file is being loaded twice.